client/components/isolated-editor/client/components/block-editor/convert-to-group-buttons.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { MenuItem } from '@wordpress/components';
+import { _x } from '@wordpress/i18n';
+import { switchToBlockType } from '@wordpress/blocks';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import { BlockSettingsMenuControls } from '@wordpress/block-editor';
+
+export function ConvertToGroupButton( {
+	onConvertToGroup,
+	onConvertFromGroup,
+	isGroupable = false,
+	isUngroupable = false,
+} ) {
+	if ( ! isGroupable && ! isUngroupable ) {
+		return null;
+	}
+
+	return (
+		<BlockSettingsMenuControls>
+			{ ( { onClose } ) => (
+				<>
+					{ isGroupable && (
+						<MenuItem
+							onClick={ () => {
+								onConvertToGroup();
+								onClose();
+							} }
+						>
+							{ _x( 'Group', 'verb' ) }
+						</MenuItem>
+					) }
+					{ isUngroupable && (
+						<MenuItem
+							onClick={ () => {
+								onConvertFromGroup();
+								onClose();
+							} }
+						>
+							{ _x(
+								'Ungroup',
```

Comment:
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?